torch/csrc/jit/passes/to_batch.cpp

```diff
@@ -0,0 +1,78 @@
+#include "torch/csrc/jit/passes/to_batch.h"
+#include "torch/csrc/jit/script/compiler.h"
+
+namespace torch { namespace jit {
+
+// map from batchTensor to {data, mask, dims}
+static std::unordered_map<Value*, std::vector<Value*>> batch_map;
+
+static void ToBatch(Block* block, Block* res_block) {
+  for (auto it = block->nodes().begin(); it != block->nodes().end(); it++) {
+    auto n = *it;
+    // replace tensor operator to BatchTensor operator
+    if(n->kind().is_aten()){
+      auto batch_graph = batch_operator_table.at(n->kind().toUnqualString());
+      WithInsertPoint guard(res_block);
+      std::vector<Value*> new_inputs;
+      for(Value *input : n->inputs()){
+        if(batch_map.find(input) != batch_map.end()){
+          auto new_input = batch_map.at(input);
+          new_inputs.insert(new_inputs.end(), new_input.begin(), new_input.end());
+        }
+        else{
```

Comment:
This seems wrong. `input` is a member of `block` but `new_inputs` are inserted into `res_block` which is in a different graph.

Reply:
`batch_map` is used to map a tensor in original graph to {data, mask, dims} in new graph. So input is a member of `block`, but `new_input` is in `res_block`. I'll add more comments for `batch_map`.

Reply:
On line 33, input is added to new_inputs. I am only concerned about the else statement.

Reply:
@zdevito Yeah, you're right. It was correct when I changed the graph in-place. And It was used to deal with non-tensor inputs for Aten operator. I'll throw a runtime error in else stmt for now and deal with it when I start handling scalarType.